core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -513,6 +523,16 @@ public SliceBuilder slice() {
         return sliceBuilder;
     }
 
+
+    public CollapseBuilder collapse() {
```

Comment:
Oups, it should yes. I pushed a commit to add the collapse builder in equals and hashcode.